tests/virt_autotest/libvirt_extend_storage_lvm.pm

```diff
@@ -0,0 +1,143 @@
+
+    record_info "LVM Storage Pool list";
+    assert_script_run "virsh pool-list --all | grep guest_image_lvm";
+
+    record_info "LVM Storage Pool start";
+    assert_script_run "virsh pool-start guest_image_lvm";
+
+    record_info "LVM Storage Pool autostart";
+    assert_script_run "virsh pool-autostart guest_image_lvm";
+
+    record_info "LVM Storage Pool info";
+    assert_script_run "virsh pool-info guest_image_lvm";
+
+    # Create a Storage Volume from LVM storage pool
+    record_info "Create a LVM Storage Volume";
+    assert_script_run("virsh vol-create-as guest_image_lvm $_-storage 1G", 120) foreach (keys %virt_autotest::common::guests);
+
+    record_info "List LVM Storage Volumes";
+    assert_script_run("ls /dev/lvm_vg", 120) foreach (keys %virt_autotest::common::guests);
+    assert_script_run("virsh vol-list guest_image_lvm | grep $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+
+    record_info "Dump LVM Storage Volumes in XML";
+    assert_script_run("virsh vol-dumpxml --pool guest_image_lvm $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+
+    # Attach a Storage Volume to guest system
+    record_info "Attached";
+    my $target = (is_xen_host) ? "xvdx" : "vdx";
+    assert_script_run("virsh attach-disk --domain $_ --source `virsh vol-path --pool guest_image_lvm $_-storage` --target $target", 120) foreach (keys %virt_autotest::common::guests);
+
+    record_info "Clone";
+    assert_script_run("virsh vol-clone --pool guest_image_lvm $_-storage $_-clone", 120) foreach (keys %virt_autotest::common::guests);
+    assert_script_run("virsh vol-info --pool guest_image_lvm $_-clone", 120) foreach (keys %virt_autotest::common::guests);
+
+    ## Cleanup
+    # Detach a Storage Volume from guest system
+    record_info "Detached";
+    assert_script_run("virsh detach-disk $_ $target", 120) foreach (keys %virt_autotest::common::guests);
+
+    # Delete and Remove a Storage Volume from LVM-based storage pool
+    record_info "Remove";
+    assert_script_run("virsh vol-delete --pool guest_image_lvm $_-clone", 120) foreach (keys %virt_autotest::common::guests);
+    assert_script_run("virsh vol-delete --pool guest_image_lvm $_-storage", 120) foreach (keys %virt_autotest::common::guests);
+
+    # Destroy the LVM volume group storage pool
+    record_info "LVM Storage Pool destroy";
+    assert_script_run "virsh pool-destroy guest_image_lvm";
+    assert_script_run "virsh pool-delete guest_image_lvm";
+    assert_script_run "virsh pool-undefine guest_image_lvm";
+    assert_script_run "virsh pool-list --all";
+    assert_script_run "vgdisplay";
+
+    # Delete a Physical Volume (PV) with LVM
+    validate_script_output("pvremove -y ${lvm_disk}1", sub { m/successfully wiped/ }, $timeout);
+    assert_script_run 'pvdisplay';
+
+    # Wipe Hard Disk Clean via dd
+    wipe_hard_disk($lvm_disk);
+}
+
+# Wipe Hard Disk Clean via dd
+sub wipe_hard_disk {
+    my $hard_disk_name = @_;
```

Comment:
the left end is a variable, while the right end is a array. it should be missing '()' around the variable or please use 'shift' instead of '@_'.

Reply:
Great! change to use use 'shift' instead of '@_' now.